packages/cli/src/auth.ts

```diff
@@ -97,3 +113,51 @@ function printMe(medplum: MedplumClient): void {
     console.log('Not logged in');
   }
 }
+
+async function medplumAuthorizationCodeLogin(medplum: MedplumClient): Promise<void> {
+  await startWebServer(medplum);
+  const loginUrl = new URL(medplum.getAuthorizeUrl());
+  loginUrl.searchParams.set('client_id', clientId);
+  loginUrl.searchParams.set('redirect_uri', redirectUri);
+  loginUrl.searchParams.set('scope', 'openid');
+  loginUrl.searchParams.set('response_type', 'code');
+  loginUrl.searchParams.set('prompt', 'login');
+  await openBrowser(loginUrl.toString());
+}
+
+async function jwtBearerLogin(medplum: MedplumClient, profile: Profile): Promise<string> {
+  const header = {
+    typ: 'JWT',
+    alg: 'HS256',
+  };
+
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+  const data = {
+    aud: `${profile.baseUrl}${profile.audience}`,
+    iss: profile.issuer,
+    sub: profile.subject,
+    nbf: currentTimestamp,
+    iat: currentTimestamp,
+    exp: currentTimestamp + 604800, // expiry time is 7 days from time of creation
+  };
+  const encodedHeader = encodeBase64(JSON.stringify(header));
+  const encodedData = encodeBase64(JSON.stringify(data));
+  const token = `${encodedHeader}.${encodedData}`;
+  const signature = createHmac('sha256', profile.clientSecret as string)
+    .update(token)
+    .digest('base64url');
+  const signedToken = `${token}.${signature}`;
+
+  const formBody = new URLSearchParams();
+  formBody.set('grant_type', 'urn:ietf:params:oauth:grant-type:jwt-bearer');
+  formBody.set('client_id', profile.clientId as string);
+  formBody.set('assertion', signedToken);
+  formBody.set('scope', profile.scope ?? '');
+
+  const res = await medplum.post(profile.tokenUrl as string, formBody.toString(), 'application/x-www-form-urlencoded', {
+    credentials: 'include',
+  });
+
+  const obj = await JSON.parse(res);
```

Comment:
Is this necessary?  `res` should be a JSON object already.

Reply:
I ran `console.log(typeof res)` and it returned string, and the active login will be empty without `JSON.parse(res)`

Here's the output without JSON.parse()

```
string
{"access_token":"{ACCESS_TOKEN}","scope":"create_users,place_orders,patient360,user/*.*","token_type":"Bearer","expires_in":604800,"hg_rest_api_version":"04-01-2023"}
```